ts/Extensions/Annotations/Types/VerticalLine.ts

```diff
@@ -58,31 +59,34 @@ class VerticalLine extends Annotation {
         target: Highcharts.AnnotationControllable
     ): Highcharts.AnnotationMockPointOptionsObject {
         var annotation = target.annotation as Highcharts.AnnotationVerticalLine,
+            chart = annotation.chart,
+            inverted = chart.inverted,
             point = annotation.points[0],
-            xy = MockPoint.pointToPixels(point, true),
-            y = xy.y,
-            offset = annotation.options.typeOptions.label.offset;
-
-        if (annotation.chart.inverted) {
-            y = xy.x;
-        }
+            left = pick(point.series.yAxis?.left, 0),
```

Comment:
I know that optional chaining is convenient, however, it generates a lot of nasty variables in JS.
Could you change that to `left = pick(point.series.yAxis && point.series.yAxis.left, 0),`

Reply:
We've already had a conversation about it a year ago and agreed that one optional chaining check is not that bad. The problem is getting bigger with more checks than one. However, I agree, we can get rid of them totally.